src/providers/hana/qgshanaproviderconnection.cpp

```diff
@@ -0,0 +1,354 @@
+    QVariant> *options ) const
+{
+  checkCapability( Capability::CreateVectorTable );
+
+  QgsDataSourceUri newUri { uri() };
+  newUri.setSchema( schema );
+  newUri.setTable( name );
+  // Set geometry column if it's not aspatial
+  if ( wkbType != QgsWkbTypes::Type::Unknown &&  wkbType != QgsWkbTypes::Type::NoGeometry )
+  {
+    newUri.setGeometryColumn( options->value( QStringLiteral( "geometryColumn" ), QStringLiteral( "geom" ) ).toString() );
+  }
+  QMap<int, int> map;
+  QString errCause;
+  QgsVectorLayerExporter::ExportError errCode = QgsHanaProvider::createEmptyLayer(
+        newUri.uri(),
+        fields,
+        wkbType,
+        srs,
+        overwrite,
+        &map,
+        &errCause,
+        options
+      );
+  if ( errCode != QgsVectorLayerExporter::ExportError::NoError )
+  {
+    throw QgsProviderConnectionException( QObject::tr( "An error occurred while creating the vector layer: %1" ).arg( errCause ) );
+  }
+}
+
+QString QgsHanaProviderConnection::tableUri( const QString &schema, const QString &name ) const
+{
+  const auto tableInfo { table( schema, name ) };
+
+  QgsDataSourceUri dsUri( uri() );
+  dsUri.setTable( name );
+  dsUri.setSchema( schema );
+  dsUri.setGeometryColumn( tableInfo .geometryColumn() );
+  return dsUri.uri( false );
+}
+
+void QgsHanaProviderConnection::dropVectorTable( const QString &schema, const QString &name ) const
+{
+  checkCapability( Capability::DropVectorTable );
+  dropTable( schema, name );
+}
+
+void QgsHanaProviderConnection::renameTable( const QString &schema, const QString &name, const QString &newName ) const
+{
+  executeSqlStatement( QStringLiteral( "RENAME TABLE %1.%2 TO %1.%3" )
+                       .arg( QgsHanaUtils::quotedIdentifier( schema ) )
+                       .arg( QgsHanaUtils::quotedIdentifier( name ) )
+                       .arg( QgsHanaUtils::quotedIdentifier( newName ) ) );
+}
+
+void QgsHanaProviderConnection::renameVectorTable( const QString &schema, const QString &name, const QString &newName ) const
+{
+  checkCapability( Capability::RenameVectorTable );
+  renameTable( schema, name, newName );
+}
+
+void QgsHanaProviderConnection::createSchema( const QString &name ) const
+{
+  checkCapability( Capability::CreateSchema );
+  executeSqlStatement( QStringLiteral( "CREATE SCHEMA %1" )
+                       .arg( QgsHanaUtils::quotedIdentifier( name ) ) );
+}
+
+void QgsHanaProviderConnection::dropSchema( const QString &name,  bool force ) const
+{
+  checkCapability( Capability::DropSchema );
+  executeSqlStatement( QStringLiteral( "DROP SCHEMA %1 %2" )
+                       .arg( QgsHanaUtils::quotedIdentifier( name ) )
+                       .arg( force ? QStringLiteral( "CASCADE" ) : QString() ) );
+}
+
+QList<QVariantList> QgsHanaProviderConnection::executeSql( const QString &sql ) const
+{
+  checkCapability( Capability::ExecuteSql );
+  if ( sql.trimmed().startsWith( "SELECT", Qt::CaseSensitivity::CaseInsensitive ) )
```

Comment:
It's probably more robust to
- prepare the statement
- retrieve the result set metadata
- check how many columns are returned

If the number of result set columns is 0, it's not a query.

Queries do not necessarily start with `SELECT`, e.g. `WITH q1 AS (SELECT * FROM t1) SELECT * FROM q1` is a query not starting with `SELECT`.

Reply:
Done in 78b6fde.